pkg/subscriptions/subscriptions.go

```diff
@@ -155,6 +155,27 @@ func getMountsMap(path string) (string, string, error) { //nolint
 	return "", "", fmt.Errorf("unable to get host and container dir from path: %s", path)
 }
 
+// Return true iff the system is in FIPS mode as determined by reading
+// /proc/sys/crypto/fips_enabled and disableFips is not true.
+func shouldAddFIPSMounts(disableFips bool) bool {
+	if disableFips {
+		return false
+	}
+
+	fips_enabled, err := os.ReadFile("/proc/sys/crypto/fips_enabled")
+	if err != nil {
+		logrus.Errorf("Failed to read /proc/sys/crypto/fips_enabled to determine FIPS state: %v", err)
+		return false
+	}
```

Comment:
This will log an error when the file does not exists, it not clear to me does the kernel always has the file or is that set to some CONFIG_ options in the kernel. If so we should certainly ignore ENOENT?

Second, this logs an error and then assumes fips is disabled, Given this is a security option should we not outright fail with the error (e.g. return the error to the caller)? I do realize MountsWithUIDGID() does not return an error so we would need to add a new function so maybe it is not worth it. Maybe it is not important enough but I always very much find it odd to log an error but then start the container anyway. Either way the issue is pre existing so I would not block this on it.

Reply:
The kernel always has that file, so there's no need to ignore ENOENT.

I can make the change to bubble up the error, but as you said, it's going to be a larger change because MountsWithUIDGID() does not currently fail. Maybe we should do that in a separate PR afterwards?

Reply:
> The kernel always has that file, so there's no need to ignore ENOENT.

@neverpanic no always. I work with systems where **CONFIG_CRYPTO_FIPS** is not always enabled.

Reply:
Should be fixed now.